### ISSUE TITLE
Libraries are deduped (differs from pkg-config)

### ISSUE THREAD

**Issue by floppym** [pkgconf/pkgconf#78]
Given 2 pc files:

```
% cat foo.pc
Name: foo
Description: test
Version: 1.0
Libs: -lfoo -ldl

% cat bar.pc
Name: bar
Description: test
Version: 1.0
Requires: foo
Libs: -lbar -ldl
```

pkgconf and pkg-config produce different output.

```
% pkg-config --libs bar
-lbar -ldl -lfoo -ldl

% pkgconf --libs bar
-lbar -lfoo -ldl
```

I discovered this while debugging a build failure for the efibootmgr software. In case you are curious to see some of the weird pkgconfig files that people create:

https://github.com/floppym/efivar/commit/e9626bda6fe09892ee5ed73931995f99829334b7

**Comment by kaniini** [comment]
The deduplication is intentional in non-static mode.  pkg-config also has nieve deduplication code, but ours is better.

I assume that efibootmgr is a static build?

**Comment by kaniini** [comment]
Actually, the fix you made seems to break an invalid pkg-config file to be honest.

@bapt what do you think?

**Comment by floppym** [comment]
> I assume that efibootmgr is a static build?

No, it is not.

> Actually, the fix you made seems to break an invalid pkg-config file to be honest.

I don't know what you mean by that.

**Comment by kaniini** [comment]
@floppym 

can you drop by `#pkgconf` on freenode so we can help determine whether this is a pkgconf bug or a bug in the file to begin with?  we need more information, as pkgconf does make optimizations that pkg-config doesn't, and generally, those optimizations depend on .pc files being correct.  I think the change you made to efivar, for example, fixes the problem and it was just dumb luck that it ever worked on freedesktop's implementation to begin with.

we would like to fix this before releasing 0.9.12 if it really winds up being a problem on our side, but bug-for-bug compatibility, in general, isn't a design goal of pkgconf.

**Comment by kaniini** [comment]
```
16:09 <@bapt> actually given you push back
16:09 <@bapt> is the pc files are properly done
16:09 <@bapt> then the dependency chain will always put things in the right order
16:10 <@bapt> so yes it is not a bug
16:10 <@bapt> it is pkgconf being pedantic on pc files which I think is a good thing
16:13 <@kaniini> i think it's a pkg-config bug
```

But we should more explicitly document that pkgconf is more pedantic.  So I will commit that documentation change soon.
